Added battery charger and backlight specs

asadate opened the pull request:
Updated power.json to add battery charger and display backlight driver specifications

asadate committed:
Added battery charger and backlight specs

Updated power.json to add battery charger and display backlight driver specifications

Carrotman42 commented on part-spec/power.json:
```diff
@@ -697,6 +697,507 @@
+        "dimmingRatioTyp": {
+          "description": "typical dimming ratio of a device",
+          "type": "string",
+          "example": "20:1"
```
Ideally, tools wouldn't have to parse strings for structured information. Should we have something like a "Ratio" type which has a numerator and denominator?

asadate replied:
Good idea. I'll work on some proposals to update the definition file and this property.

Carrotman42 commented on part-spec/power.json:
```diff
@@ -697,6 +697,507 @@
+          "description": "whether bc 1.2 detection is built in",
+          "type": "boolean"
+        },
+        "TCPCSupport": {
```
it's a bit weird that `usb` (and `pd` below) aren't capitalized but `TCPC` (and `BC` above) is. Seems like we should be consistent.

asadate replied:
Agreed. I'll make the change.

Carrotman42 reviewed (changes requested):
Apologies for the delay, this totally slipped off my radar.

asadate committed:
Update power.json - Changed ratio type from string to object

Addressed feedback from review - Changes "xRatio" property type from string to object

asadate committed:
Updated examples.md

Added examples for the LDO, Load Switch, Display Backlight Driver.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          "description": "common component identifying information, such as mpn",
+          "$ref": "definitions.json#/componentID"
+        },
+        "vinTyp": {
```
would it make sense to combine Typ, Min, Max into an object for reuse since we see this pattern being used often?

asadate replied:
No strong preference from my side. Whatever works better for tools developers.  My proposal is to release this as is for the rev 0.8 but to come back in rev 0.9 and make the change for all  the components we've defined so far.

rsnancollas replied:
Yeah I think it makes sense to push this to v0.9.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          "comment": "units of percentage",
+          "$ref": "definitions.json#/unit"
+        },
+        "integratedBoost": {
```
some overlap with integrated fets here, but I guess this could be just the switching drivers

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          }
+        },
+        "package": {
+          "description": "package size of resistor",
```
remove "resistor"

asadate replied:
Good catch. Will remove.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+            "buck",
+            "boost",
+            "buck-boost",
+            "other"
```
since these are examples, could remove "other"

asadate replied:
Will remove

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          "type": "string"
+        },
+        "batteryConfig": {
+          "description": "battery configuration supported by the device",
```
this should be an array since chargers can support multiple batter configs

rsnancollas replied:
Also the number in series is generally more important than the number in parallel as I recall

asadate replied:
Yes, I'll change this and couple of other property types to array since multiple configs can be supported. As for the S & P, wanted to broaden it out so that it's not too centric to Notebooks.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          ],
+          "type": "string"
+        },
+        "inputPowerSource": {
```
Would this just be more of a voltage range + some additional info if USB PD protocol is supported?

asadate replied:
This is more about the type of inputs supported: e.g: AC adapter, USB C with PD protocol, USB 5V or USB 20V...

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          "comment": "units of %",
+          "$ref": "definitions.json#/unit"
+        },
+        "batteryChargeVoltageTyp": {
```
couldn't this be variable - does a typical value make sense?

asadate replied:
Let's discuss more. I've mostly seen this defined as a value.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,516 @@
+          "$ref": "definitions.json#/unit"
+        },
+        "efficiency": {
+          "description": "charge efficiency vs charge current of a device",
```
this might be a series of graphs depending on Vin/Vout

asadate replied:
I think it's ok because the series of graph option should already be covered by the way we defined "graph".

asadate committed:
Update power.json

Addressed review feedback

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+          {"numerator": 20},
+          {"denominator": 1}
+          ]
+        },
```
I don't think this syntax is legal. Instead, I'd do:

"description": "typical dimming ratio of a device",
"$ref": "definitions.json#/ratio"


And add the following definition to definitions.json:

  "ratio": {
    "type": "object",
    "properties": {
      "numerator": {
        "description": "numerator of ratio",
        "type": "number"
      },
      "denominator": {
        "description": "denominator of ratio",
        "type": "number"
      }
    }
  },

asadate replied:
changed in power file. Will also change the definitions file

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+          {"numerator": 20},
+          {"denominator": 1}
+          ]
+        },
```
I don't think this syntax is legal. Instead, I'd do:

"description": "typical dimming ratio of a device",
"$ref": "definitions.json#/ratio"


And add the following definition to definitions.json:

  "ratio": {
    "type": "object",
    "properties": {
      "numerator": {
        "description": "numerator of ratio",
        "type": "number"
      },
      "denominator": {
        "description": "denominator of ratio",
        "type": "number"
      }
    }
  },

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+        },
+        "dimmingRatioMin": {
+          "description": "minimum dimming ratio of a device",
+          "type": "object",
```
same comment as above: use ratio

asadate replied:
changed in power file. Will also change the definitions file

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+          "example": [
+          {"numerator": 40},
+          {"denominator": 1}
+          ]
```
same as above, ratio

asadate replied:
changed in power file. Will also change the definitions file.

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+            "nvdc",
+            "hpb"
+          ],
+          "type": "array"
```
should be:

"type": "string"

asadate replied:
done

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+            "4S",
+            "4S2P"
+          ],
+          "type": "array"
```
"type": "array",
"items": {
        "type": "string"
}

asadate replied:
done

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+            "Li-Ion",
+            "LiPo"
+          ],
+          "type": "array"
```
"type": "array",
"items": {
        "type": "string"
}

asadate replied:
changed

rsnancollas commented on part-spec/power.json:
```diff
@@ -697,6 +697,515 @@
+          "description": "component's package size and description",
+          "items": {
+            "$ref": "definitions.json#/package"
+          }
```
not "items", just:

"$ref": "definitions.json#/package"

asadate replied:
changed and also made the change for other components where package is called,

asadate committed:
Update power.json

Made changes per review. Also fixed the syntax for the package definition in other components in the file.

asadate committed:
Update definitions.json

Added the "ratio" object definition.

rsnancollas committed:
Merge branch 'main' into power4
